File: MetaGeta/MetaGeta.GUI/Editors/StringListEditor.vb
Namespace Editors
    Public Class StringListEditor
        Inherits ItemListEditor

        Private m_DialogCaption, m_DialogPrompt As String

        Public Property DialogCaption() As String
            Get
                Return m_DialogCaption
            End Get
            Set(ByVal value As String)
                m_DialogCaption = value
            End Set
        End Property

        Public Property DialogPrompt() As String
            Get
                Return m_DialogPrompt
            End Get
            Set(ByVal value As String)
                m_DialogPrompt = value
            End Set
        End Property

        Protected Overrides Function CreateItem() As String
            Dim prompt As New StringInputPrompt(DialogCaption, DialogPrompt)
            prompt.Owner = Window.GetWindow(Me)
            Dim result = prompt.ShowDialog()
            If result.HasValue AndAlso result.Value Then
                Return prompt.EditingString
            Else
                Return Nothing
            End If
        End Function
    End Class
End Namespace

File: MetaGeta/MetaGeta.GUI/Editors/StringListEditor.cs
// Copyright 2009 Will Thomas
// 
// This file is part of MetaGeta.
// 
// MetaGeta is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// MetaGeta is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with MetaGeta. If not, see <http://www.gnu.org/licenses/>.

#region

using System.Windows;

#endregion

namespace MetaGeta.GUI.Editors {
    public class StringListEditor : ItemListEditor {
        private string m_DialogCaption;

        private string m_DialogPrompt;

        public string DialogCaption {
            get { return m_DialogCaption; }
            set { m_DialogCaption = value; }
        }

        public string DialogPrompt {
            get { return m_DialogPrompt; }
            set { m_DialogPrompt = value; }
        }

        protected override string CreateItem() {
            var prompt = new StringInputPrompt(DialogCaption, DialogPrompt);
            prompt.Owner = Window.GetWindow(this);
            bool? result = prompt.ShowDialog();
            if (result.HasValue && result.Value)
                return prompt.EditingString;
            else
                return null;
        }
    }
}